src/support/utils/casting.ts

```
@@ -0,0 +1,65 @@
+  if (typeof input === "number") return input;
+
+  // Check if the input is a string that can be converted to a number
+  if (typeof input === "string") {
+    // Trim leading and trailing whitespaces
+    const trimmedInput = input.trim();
+
+    // Check if the trimmed string is a numeric value
+    const num = Number(trimmedInput);
+
+    // Check if the result is a valid number and not NaN
+    if (!isNaN(num)) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)